packages/ui-causal-graph-editor/src/shared/serializer.tsx

```diff
@@ -49,7 +49,11 @@ function removeNodePrefix<T extends keyof FlatNodeRenderingMeta>(key: T): keyof
  * @param source optional source to include in output data
  * @param destination optional destination to include in output data
  */
-export function serializeGraphEdge(attributes: SimulationEdge, source?: string, destination?: string): CausalGraphEdge {
+export function serializeGraphEdge(
+    attributes: SimulationEdge,
+    source?: CausalGraphNode,
+    destination?: CausalGraphNode
```

Comment:
Are those still optional? If we're expecting those to get passed everywhere I'd make them non-optional so typescript will yell at us if we miss it somewhere

Reply:
Marked them as required here and in the type definition